src/dune_engine/action_builder.mli

```diff
@@ -112,9 +112,7 @@ val dep_on_alias_if_exists : Alias.t -> bool t
 val dep_on_alias_rec :
   Alias.Name.t -> Context_name.t -> Source_tree.Dir.t -> bool t
 
-(** Compute the set of source of all files present in the sub-tree starting at
-    [dir] and record them as dependencies. *)
-val source_tree : dir:Path.t -> Path.Set.t t
+val memo_deps : (Dep.Set.t * 'a) Memo.t -> 'a t
```

Comment:
I'm not entirely sure what's a good name for this function.

Reply:
Looks like we use `dyn_` prefix for this sort of stuff -- see below, e.g. `dyn_path_set`.

Reply:
Can you add a comment?

Reply:
Done.